builder/dockerfile/dispatchers_windows.go

```diff
@@ -17,8 +17,28 @@ func normaliseWorkdir(current string, requested string) (string, error) {
 		return "", fmt.Errorf("cannot normalise nothing")
 	}
 
-	current = filepath.FromSlash(current)
-	requested = filepath.FromSlash(requested)
+	// `filepath.Clean` will replace "" with "." so skip in that case
+	if current != "" {
+		current = filepath.Clean(current)
+	}
+	if requested != "" {
+		requested = filepath.Clean(requested)
+	}
+
+	// If either current or requested in Windows is:
+	// C:
+	// C:.
+	// then an error will be thrown as the definition for the above
+	// refers to `current directory on drive C:`
+	// Since filepath.Clean() will automatically normalize the above
+	// to `C:.`, we only need to check the last format
+	pattern := regexp.MustCompile(`^[a-zA-Z]:\.$`)
```

Comment:
Thanks @tiborvass. The PR has been updated. Please let me know if there are any issues.